persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/SignificantlyMutatedGeneSparkRepository.java

```diff
@@ -0,0 +1,80 @@
+package org.cbioportal.persistence.spark;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.cbioportal.model.MutSig;
+import org.cbioportal.model.meta.BaseMeta;
+import org.cbioportal.persistence.SignificantlyMutatedGeneRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@Component
+@Qualifier("significantlyMutatedGeneSparkRepository")
+public class SignificantlyMutatedGeneSparkRepository implements SignificantlyMutatedGeneRepository {
+
+    @Autowired
+    private SparkSession spark;
+
+    @Value("${data.parquet.folder}")
+    private String PARQUET_DIR;
+
+    private static Log logger = LogFactory.getLog(SignificantlyMutatedGeneSparkRepository.class);
+
+    @Override
+    public List<MutSig> getSignificantlyMutatedGenes(String studyId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        System.out.println(PARQUET_DIR);
+        Dataset<Row> mutationDf = spark.read()
+            .parquet(PARQUET_DIR + "/" + studyId + "/data_mutations_extended.txt.parquet");
+
+        // for MSK
+        if (studyId.contains("msk")) {
+            Dataset<Row> mutationsMskcc = spark.read()
+                .parquet(PARQUET_DIR + "/" + studyId + "/data_mutations_mskcc.txt.parquet");
+            mutationsMskcc = mutationsMskcc.drop("cDNA_change");
```

Comment:
Why is this here?

Reply:
drop() is called before unionByName() because union requires both Datasets to have the same column names.